### DISCUSSION TITLE
authenticated uploading ok, not listing files in a bucket

### DISCUSSION THREAD

**Discussion by hududed** [supabase/supabase#18731] [Questions]
So from what I gather issues with storage usually revolves around policies but since it's not clear how policies should be written, I just used the template for all operations:

```
CREATE POLICY "Enable insert for authenticated users only" ON "storage"."buckets"
AS PERMISSIVE FOR ALL
TO authenticated

WITH CHECK (true)
```

the running `supabase_client.storage.from_(bucket_name).upload(new_file_name, file_content)` works fine.

but this `files = supabase_client.storage.from_(bucket_name).list()` doesnt work.

**Comment by GaryAustin1** [comment]
You should use the guide templates as a starter.

But the problem with your current policy is it does not  have a USING clause.   If the policy is for select/update or ALL then you need USING (true).   WITH CHECK is optional for ALL and will default to the USING clause.

**Reply by hududed** [reply]
Ah ok. so just to confirm:

```
CREATE POLICY "Enable all operations for authenticated users only" ON "storage"."buckets"
AS PERMISSIVE FOR ALL
TO authenticated
USING (true)
WITH CHECK (true)
```

should work for lists and downloads?

**Reply by GaryAustin1** [reply]
I'm a bit confused now.   Your policy is on storage.buckets.   In order for upload to work the policy would have to be on storage.objects.
To list files you need SELECT access to storage.objects (not buckets).

So changing the buckets to objects...
```sql
CREATE POLICY "Enable all operations for authenticated users only" ON "storage"."objects"
AS PERMISSIVE FOR ALL
TO authenticated
USING (true)
WITH CHECK (true)
```
Would allow signed in users to do any operation to all files in all buckets.  That is a very broad policy, but should work.
